concrete/controllers/element/dashboard/marketplace/extend.php

```diff
@@ -62,7 +62,7 @@ private function getPurchaseConnectionUrl(Connection $connection, string $url):
     public function view()
     {
         $connection = $this->packageRepository->getConnection();
-        if ($this->packageRepository->validate($connection)) {
+        if ($connection && $this->packageRepository->validate($connection)) {
```

Comment:
My suggestion is a slightly more restrictive condition. If you happen to accept this version, remember to add `use Concrete\Core\Marketplace\ConnectionInterface` at the top of the file.

```suggestion
        if ($connection instanceof ConnectionInterface && $this->packageRepository->validate($connection)) {
```